src/Ledger/Epoch.agda

```diff
@@ -33,16 +34,51 @@ record EpochStructure : Set₁ where
         sucᵉ : Epoch → Epoch
         instance DecEq-Epoch : DecEq Epoch
 
+  _>ˢ_ : Slot → Slot → Set
+  _>ˢ_ = flip _<ˢ_
+
   _≥ˢ_ : Slot → Slot → Set
-  _≥ˢ_ = ¬_ ∘₂ _<ˢ_
+  x ≥ˢ y = (x >ˢ y) ⊎ (x ≡ y)
 
   _≤ˢ_ : Slot → Slot → Set
-  _≤ˢ_ = flip _≥ˢ_
+  x ≤ˢ y = (x <ˢ y) ⊎ (x ≡ y)
+
+  tri : Trichotomous _≡_ _<ˢ_
+  tri = IsStrictTotalOrder.compare Slot-STO
+
+  ≤ˢ⇒¬>ˢ-mp : {x y : Slot} → x ≤ˢ y → Tri (x <ˢ y) (x ≡ y) (x >ˢ y) → ¬ (x >ˢ y)
+  ≤ˢ⇒¬>ˢ-mp _           (tri< _     _     ¬x>y)  x>y  = ¬x>y x>y
+  ≤ˢ⇒¬>ˢ-mp _           (tri≈ _     _     ¬x>y)  x>y  = ¬x>y x>y
+  ≤ˢ⇒¬>ˢ-mp (inj₁ x<y)  (tri> ¬x<y  _     _)     _    = ¬x<y x<y
+  ≤ˢ⇒¬>ˢ-mp (inj₂ x≡y)  (tri> _     ¬x≡y  _)     _    = ¬x≡y x≡y
+
+  ≤ˢ⇒¬>ˢ : {x y : Slot} → x ≤ˢ y → ¬ (x >ˢ y)
+  ≤ˢ⇒¬>ˢ x≤y x>y = ≤ˢ⇒¬>ˢ-mp x≤y (tri _ _) x>y
 
   open IsStrictTotalOrder Slot-STO using () renaming (_<?_ to _<ˢ?_) public
 
+  open StrictToNonStrict _≡_ _<ˢ_
+
+  ≤ˢ-isDecTotalOrder : IsDecTotalOrder _≡_ _≤ˢ_
+  ≤ˢ-isDecTotalOrder = isDecTotalOrder Slot-STO
+
+  ≤ˢ-isTotalOrder : IsTotalOrder _≡_ _≤ˢ_
+  ≤ˢ-isTotalOrder = isTotalOrder Slot-STO
+
+  ≤ˢ-isPartialOrder : IsPartialOrder _≡_ _≤ˢ_
+  ≤ˢ-isPartialOrder = IsTotalOrder.isPartialOrder ≤ˢ-isTotalOrder
+
+  ≤ˢ-isPreorder : IsPreorder _≡_ _≤ˢ_
+  ≤ˢ-isPreorder = IsPartialOrder.isPreorder ≤ˢ-isPartialOrder
+
+  ≤ˢ-isTransitive : Transitive _≤ˢ_
+  ≤ˢ-isTransitive = IsPreorder.trans ≤ˢ-isPreorder
+
+  ≤ˢ-isAntisymmetric : Antisymmetric _≡_ _≤ˢ_
+  ≤ˢ-isAntisymmetric = IsPartialOrder.antisym ≤ˢ-isPartialOrder
+
```

Comment:
These are included because some of them are convenient below and others may be convenient in the future.